lib/chef/resource/chef_client_config.rb

```diff
@@ -250,20 +250,21 @@ def string_to_symbol(prop_val)
         introduced: "17.8"
 
       action :create, description: "Create a client.rb config file for configuring #{ChefUtils::Dist::Infra::PRODUCT}." do
-        unless ::Dir.exist?(new_resource.config_directory)
-          directory new_resource.config_directory do
+        [
+          new_resource.config_directory,
+          (::File.dirname(new_resource.log_location) unless new_resource.log_location.nil?),
+          new_resource.file_backup_path,
+          new_resource.file_cache_path,
+          ::File.join(new_resource.config_directory, "client.d"),
+          (::File.dirname(new_resource.pid_file) unless new_resource.pid_file.nil?),
+        ].each do |dir_path|
+          next if dir_path.nil?
+          next if ::Dir.exist?(dir_path)
+
+          directory dir_path do
             user new_resource.user unless new_resource.user.nil?
             group new_resource.group unless new_resource.group.nil?
-            mode "0750"
-            recursive true
-          end
-        end
-
-        unless ::Dir.exist?(::File.join(new_resource.config_directory, "client.d"))
-          directory ::File.join(new_resource.config_directory, "client.d") do
-            user new_resource.user unless new_resource.user.nil?
-            group new_resource.group unless new_resource.group.nil?
-            mode "0750"
+            mode dir_path == ::File&.dirname(new_resource.log_location) ? "0755" : "0750"
```

Comment:
The `&.` here could be `.` since `::File` will never be nil

Reply:
removed &